action/rlp_tx.go

```diff
@@ -83,3 +82,68 @@ func DecodeRawTx(rawData string, chainID uint32) (tx *types.Transaction, sig []b
 	pubkey, err = crypto.RecoverPubkey(rawHash[:], sig)
 	return
 }
+
+// DecodeEtherTx decodes raw data string into eth tx
+func DecodeEtherTx(rawData string) (*types.Transaction, iotextypes.Encoding, []byte, crypto.PublicKey, error) {
+	var (
+		tx       types.Transaction
+		encoding iotextypes.Encoding
+		sig      []byte
+		pubkey   crypto.PublicKey
+	)
+	//remove Hex prefix and decode string to byte
+	rawData = strings.Replace(rawData, "0x", "", -1)
+	rawData = strings.Replace(rawData, "0X", "", -1)
+	rawTxBytes, err := hex.DecodeString(rawData)
+	if err != nil {
+		return &tx, encoding, sig, pubkey, err
```

Comment:
`return nil, encoding, sig, pubkey, err`

Reply:
why want to return `nil` for tx specifically? we are returning the default value of these types here, and there's `err` to be checked

Reply:
If there was an error, the return value of non-nil `tx` just brings up some confusion to the caller side

Reply:
fixed